mitiq/interface/conversions.py

```diff
@@ -296,68 +297,83 @@ def new_scaling_function(
-            _transform_registers(scaled_circuit, new_qregs=circuit.qregs)
-            _remove_identity_from_idle(scaled_circuit, idle_qubits)
-            if circuit.cregs and not scaled_circuit.cregs:
-                scaled_circuit.add_register(*circuit.cregs)
-
-            for q, c in _measurement_order(circuit):
-                scaled_circuit.measure(q, c)
-
-        return scaled_circuit
-
-    return new_scaling_function
+            out_circuits = [out_circuit]
+
+        circuits_to_return = []
+        for out_circuit in out_circuits:
+            # Post atomic conversion
+            # PyQuil: Restore declarations, measurements, and metadata.
+            if "pyquil" in out_circuit.__module__:
+                from pyquil import Program
+                from pyquil.quilbase import Declare, Measurement
+
+                circuit = cast(Program, circuit)
+
+                # Grab all measurements from the input circuit.
+                measurements = [
+                    instr
+                    for instr in circuit.instructions
+                    if isinstance(instr, Measurement)
+                ]
+
+                # Remove memory declarations added from Cirq-pyQuil conversion.
+                new_declarations = {
+                    k: v
+                    for k, v in out_circuit.declarations.items()
+                    if k == "ro" or v.memory_type != "BIT"
+                }
+                new_declarations.update(circuit.declarations)
+
+                # Delete all declarations and measurements.
+                instructions = [
+                    instr
+                    for instr in out_circuit.instructions
+                    if not (isinstance(instr, (Declare, Measurement)))
+                ]
+
+                # Add back original declarations and measurements.
+                out_circuit = Program(
+                    list(new_declarations.values())
+                    + instructions
+                    + measurements
+                )
+
+                # Set the number of shots to the input circuit.
+                out_circuit.num_shots = circuit.num_shots
+
+            # Qiskit: Keep the same register structure and measurement order.
+            if "qiskit" in out_circuit.__module__:
+                from mitiq.interface.mitiq_qiskit.conversions import (
+                    _measurement_order,
+                    _remove_identity_from_idle,
+                    _transform_registers,
+                )
+
+                out_circuit.remove_final_measurements()
+                _transform_registers(out_circuit, new_qregs=circuit.qregs)
+                _remove_identity_from_idle(out_circuit, idle_qubits)
+                if circuit.cregs and not out_circuit.cregs:
+                    out_circuit.add_register(*circuit.cregs)
+
+                for q, c in _measurement_order(circuit):
+                    out_circuit.measure(q, c)
+            circuits_to_return.append(out_circuit)
```

Comment:
FYI: this block is unchanged apart from re-naming a variable. For some reason Git Hub diff shows it as new code block (green) but it's actually not new.

Reply:
I think GitHub is displaying it this way as not only has the variable been renamed, but the code has been added inside an `if` block (hence indenting all of this code by an additional level).